### PR TITLE
Dev/ep2019 3.6

### PR DESCRIPTION
**Pull request by fpierfed**
Port to 3.6 (latest with changes for Travis)

### CODE REVIEW

File: assopy/admin.py [hunk 1]
```diff
@@ -727,13 +727,13 @@ def do_csv_invoices(self, request, queryset):
                 'Billing notes')
 
         def e(d):
-            for k, v in d.items():
+            for k, v in list(d.items()):
```
Comment (umgelurgel) [inline assopy/admin.py]: I think the 2to3 has converted some iterators to lists unnecessarily - if they're being looped over, they might as well stay iterators.

File: assopy/forms.py [hunk 2]
```diff
@@ -230,7 +230,7 @@ def clean(self):
         data = self.cleaned_data
         o = []
         total = 0
-        for k, q in data.items():
+        for k, q in list(data.items()):
```
Comment (umgelurgel) [inline assopy/forms.py]: Unnecessary `list` conversion.

File: assopy/stats.py [hunk 3]
```diff
@@ -60,7 +60,7 @@ def movimento_cassa(**kw):
         'out': 0,
         'errors': 0,
     }
-    for k, v in output.items():
+    for k, v in list(output.items()):
```
Comment (umgelurgel) [inline assopy/stats.py]: Unnecessary `list` conversion, here and throughout the file.

File: assopy/views.py [hunk 4]
```diff
@@ -200,7 +200,7 @@ def paypal_cc_billing(request, code):
         "country": o.country,
         "address_name": o.card_name,
     }
-    qparms = urllib.urlencode([ (k,x.encode('utf-8') if isinstance(x, unicode) else x) for k,x in cc_data.items() ])
+    qparms = urllib.parse.urlencode([ (k,x.encode('utf-8') if isinstance(x, str) else x) for k,x in list(cc_data.items()) ])
```
Comment (umgelurgel) [inline assopy/views.py]: Unnecessary `list` conversion.

File: conference/admin.py [hunk 5]
```diff
@@ -349,7 +349,7 @@ def __new__(mcs, name, bases, attrs):
         if not model:
             return new_class
 
-        for name, f in model.__dict__.items():
+        for name, f in list(model.__dict__.items()):
```
Comment (umgelurgel) [inline conference/admin.py]: Unnecessary `list` conversion.

File: conference/admin.py [hunk 6]
```diff
@@ -830,7 +828,7 @@ class SplitEventForm(forms.Form):
                         etracks = set(models.EventTrack.objects\
                             .filter(event=ev)\
                             .values_list('track__track', flat=True))
-                        for sid, tracks in tmap.items():
+                        for sid, tracks in list(tmap.items()):
```
Comment (umgelurgel) [inline conference/admin.py]: Unnecessary `list` conversion.

Comment (umgelurgel) [reply conference/admin.py]: And across this file.

File: conference/users.py [hunk 7]
```diff
@@ -14,4 +14,4 @@
 def generate_random_username():
     """Returns random username of length set by RANDOM_USERNAME_LENGTH"""
     # TODO: this will break on python3.
-    return os.urandom(100).encode('hex')[:RANDOM_USERNAME_LENGTH]
+    return os.urandom(100).hex()[:RANDOM_USERNAME_LENGTH]
```
Comment (umgelurgel) [inline conference/users.py:16]: If it works, it's probably safe to remove the TODO above it :)

File: pycon/settings.py [hunk 8]
```diff
@@ -1231,15 +1231,15 @@ def cron_cleanup():
     if not DEBUG:
         raise RuntimeError('SECURE_MEDIA_ROOT cannot be a subdir of MEDIA_ROOT')
     else:
-        print
+        print()
         'WARN, SECURE_MEDIA_ROOT is a subdir of MEDIA_ROOT'
```
Comment (umgelurgel) [inline pycon/settings.py]: I think this is what should be printed.

File: pycon/settings.py [hunk 9]
```diff
@@ -1231,15 +1231,15 @@ def cron_cleanup():
...
 
 
 if not SECRET_KEY:
     if not DEBUG:
         raise RuntimeError('SECRET_KEY not set')
     else:
-        print
+        print()
         'WARN, SECRET_KEY not set'
```
Comment (umgelurgel) [inline pycon/settings.py]: I think this is what should be printed.

File: conference/models.py [hunk 10]
```diff
@@ -301,7 +301,7 @@ def findSlugForUser(self, user):
                     # The protection from slug like "str-str-str"
                     last = 0
                 continue
-            if counter > last:
+            if last is None or counter > last:
```
Comment (umgelurgel) [inline conference/models.py:304]: Why was this change needed?

File: conference/fares.py [hunk 11]
```diff
@@ -77,7 +77,8 @@ def create_fare_for_conference(code, conference, price,
     assert is_fare_code_valid(code)
     assert isinstance(conference, str), "conference should be a string"
     assert isinstance(vat_rate, Vat)
-    assert start_validity <= end_validity
+    if start_validity is not None and end_validity is not None:
```
Comment (umgelurgel) [inline conference/fares.py:80]: Why was this change needed?

File: .travis.yml [hunk 12]
```diff
@@ -10,7 +10,7 @@ git:
   depth: 1
 
 python:
-  - "2.7"
+  - "3.6"
```
Comment (umgelurgel) [inline .travis.yml:13]: 👍

**Review by umgelurgel — CHANGES_REQUESTED** [review]
Looks good overall - there's a few questions about some of the changes and it seems that the `2to3` script was a bit too eager casting things to lists in some places.

File: conference/models.py (hunk 10 above)
Comment (fpierfed) [reply conference/models.py:304]: This and the function is called with last=None which breaks the comparison.

File: conference/fares.py (hunk 11 above)
Comment (fpierfed) [reply conference/fares.py:80]: This and the function is called with start_validity =None and end_validity=None which breaks the comparison.

File: pycon/settings.py (hunk 8 above)
Comment (fpierfed) [reply pycon/settings.py]: Fixed

File: pycon/settings.py (hunk 9 above)
Comment (fpierfed) [reply pycon/settings.py]: Fixed

File: assopy/admin.py (hunk 1 above)
Comment (fpierfed) [reply assopy/admin.py]: Fixed

File: assopy/forms.py (hunk 2 above)
Comment (fpierfed) [reply assopy/forms.py]: Fixed

File: assopy/stats.py (hunk 3 above)
Comment (fpierfed) [reply assopy/stats.py]: Fixed

File: assopy/views.py (hunk 4 above)
Comment (fpierfed) [reply assopy/views.py]: Fixed

File: conference/admin.py (hunk 5 above)
Comment (fpierfed) [reply conference/admin.py]: Fixed

File: conference/admin.py (hunk 6 above)
Comment (fpierfed) [reply conference/admin.py]: Fixed

File: conference/users.py (hunk 7 above)
Comment (fpierfed) [reply conference/users.py:16]: Removed

File: assopy/middleware.py [hunk 13]
```diff
@@ -15,33 +15,33 @@ def dump_exc_plus(exclude=None):
...
-        print >>s, "Frame %s in %s at line %s" % (
+        print(file=s)
+        print("Frame %s in %s at line %s" % (
             frame.f_code.co_name,
             frame.f_code.co_filename,
-            frame.f_lineno)
+            frame.f_lineno), file=s)
 
-        for key, value in frame.f_locals.items():
+        for key, value in list(frame.f_locals.items()):
```
Comment (umgelurgel) [inline assopy/middleware.py]: Unnecessary `list`.

File: conference/fares.py [hunk 14]
```diff
@@ -116,7 +117,7 @@ def pre_create_typical_fares_for_conference(conference, vat_rate,
                                             print_output=False):
     fares = []
 
-    for fare_code in AVAILABLE_FARE_CODES.keys():
+    for fare_code in list(AVAILABLE_FARE_CODES.keys()):
```
Comment (umgelurgel) [inline conference/fares.py]: Unnecessary `list`.

File: conference/ical.py [hunk 15]
```diff
@@ -34,20 +34,20 @@ def encode(line):
 def content(name, value, params=None):
     if params:
         t = []
-        for pname, pvalue in params.items():
+        for pname, pvalue in list(params.items()):
```
Comment (umgelurgel) [inline conference/ical.py]: Unnecessary `list`.

File: conference/ical.py [hunk 16]
```diff
@@ -98,7 +98,7 @@ def __init__(self, name, *args, **kwargs):
 
     def encode(self):
         yield content('BEGIN', self.name)
-        for name, prop in self.items():
+        for name, prop in list(self.items()):
```
Comment (umgelurgel) [inline conference/ical.py]: Unnecessary `list`.

File: conference/models.py [hunk 17]
```diff
@@ -1000,7 +1000,7 @@ def events_score_by_attendance(self, conference):
         # Parallel obviously can not participate in both, so the
         # his vote should be scaled
         scores = defaultdict(lambda: 0.0)
-        for evt, users in events.items():
+        for evt, users in list(events.items()):
```
Comment (umgelurgel) [inline conference/models.py]: Unnecessary `list`.

File: conference/templatetags/conference.py [hunk 18]
```diff
@@ -237,7 +237,7 @@ def schedule_context(schedule):
 
     TIME_STEP = 15
     dbtracks = dict((t.track, (ix, t)) for ix, t in enumerate(schedule.track_set.all()))
-    _itracks = dict(dbtracks.values())
+    _itracks = dict(list(dbtracks.values()))
```
Comment (umgelurgel) [inline conference/templatetags/conference.py]: Unnecessary `list`.

File: conference/templatetags/conference.py [hunk 19]
```diff
@@ -539,7 +539,7 @@ def overbooked_events(context, conference):
     c = _request_cache(context['request'], 'schedules_overbook')
     if not c:
         data = models.Schedule.objects.expected_attendance(conference)
-        for k, v in data.items():
+        for k, v in list(data.items()):
```
Comment (umgelurgel) [inline conference/templatetags/conference.py]: Unnecessary `list`.

File: conference/templatetags/conference.py [hunk 20]
```diff
@@ -977,7 +977,7 @@ def voting_data(conference):
         for tid, type, language in results:
             groups[type][language].append(tid)
 
-    for k, v in groups.items():
+    for k, v in list(groups.items()):
```
Comment (umgelurgel) [inline conference/templatetags/conference.py]: Unnecessary `list`.

File: conference/templatetags/conference.py [hunk 21]
```diff
@@ -1568,7 +1568,7 @@ def conference_js_data(tags=None):
 
     cts = dict(ContentType.objects.all().values_list('id', 'model'))
     items = {}
-    for t, objects in tags.items():
+    for t, objects in list(tags.items()):
```
Comment (umgelurgel) [inline conference/templatetags/conference.py]: Unnecessary `list`.

File: conference/utils.py [hunk 22]
```diff
@@ -72,9 +72,9 @@ def _input_for_ranking_of_talks(talks, missing_vote=5):
     for vote in votes:
         users[vote.user_id][vote.vote].append(vote.talk_id)
 
-    for votes in users.values():
+    for votes in list(users.values()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 23]
```diff
@@ -169,7 +169,7 @@ def addEvents(self, events):
 
     def removeEventsByTag(self, *tags):
         tags = set(tags)
-        for events in self.events.values():
+        for events in list(self.events.values()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 24]
```diff
@@ -209,7 +209,7 @@ def fromTracks(cls, tids):
         tracks = set(Track.objects\
             .filter(id__in=tids)\
             .values_list('track', flat=True))
-        for t in tt.events.keys():
+        for t in list(tt.events.keys()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 25]
```diff
@@ -289,7 +289,7 @@ def iterOnTimes(self, step=None):
         """
         self._analyze()
         trasposed = defaultdict(list)
-        for events in self.events.values():
+        for events in list(self.events.values()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 26]
```diff
@@ -308,7 +308,7 @@ def limits(self):
         Returns start date and the end of the TimeTable
         """
         start = end = None
-        for e in self.events.values():
+        for e in list(self.events.values()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 27]
```diff
@@ -341,7 +341,7 @@ def slice(self, start=None, end=None):
                 end = None
         events = dict(self.events)
         if start or end:
-            for track, evs in events.items():
+            for track, evs in list(events.items()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 28]
```diff
@@ -359,23 +359,23 @@ def adjustTimes(self, start=None, end=None):
...
+            'tracks': list(self.events.keys()),
             'tags': set(),
             'talk': None,
             'time': None,
             'duration': None,
         }
         e0, e1 = self.limits()
         if start and e0 and start < e0.time():
-            for track, events in self.events.items():
+            for track, events in list(self.events.items()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 29]
```diff
@@ -359,23 +359,23 @@ def adjustTimes(self, start=None, end=None):
...
                 e = dict(tpl)
                 e['time'] = datetime.combine(events[0]['time'].date(), start)
                 e['duration'] = (e0 - e['time']).seconds / 60
                 self.events[track].insert(0, e)
 
         if end and e1 and end > e1.time():
             d = (datetime.combine(date.today(), end) - e1).seconds / 60
-            for track, events in self.events.items():
+            for track, events in list(self.events.items()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/utils.py [hunk 30]
```diff
@@ -592,7 +592,7 @@ def changesAtTime(self, start):
         Returns the rows that introduce a change in the past tense.
         """
         output = []
-        for key, item in self._data.items():
+        for key, item in list(self._data.items()):
```
Comment (umgelurgel) [inline conference/utils.py]: Unnecessary `list`.

File: conference/views.py [hunk 31]
```diff
@@ -434,7 +434,7 @@ def voting(request):
             return http.HttpResponseBadRequest('anonymous user not allowed')
 
         data = dict((x.id, x) for x in talks)
-        for k, v in filter(lambda x: x[0].startswith('vote-'), request.POST.items()):
+        for k, v in [x for x in list(request.POST.items()) if x[0].startswith('vote-')]:
```
Comment (umgelurgel) [inline conference/views.py]: Unnecessary `list`.

File: conference/views.py [hunk 32]
```diff
@@ -540,7 +540,7 @@ class OptionForm(forms.Form):
             # as a filter to those associated with talk.
             allowed = set()
             ctt = ContentType.objects.get_for_model(models.Talk)
-            for t, usage in dataaccess.tags().items():
+            for t, usage in list(dataaccess.tags().items()):
```
Comment (umgelurgel) [inline conference/views.py]: Unnecessary `list`.

File: conference/views.py [hunk 33]
```diff
@@ -647,7 +647,7 @@ def covers(request, conference):
     from collections import defaultdict
     tracks = defaultdict(dict)
     for s in schedules:
-        for t in s['tracks'].values():
+        for t in list(s['tracks'].values()):
```
Comment (umgelurgel) [inline conference/views.py]: Unnecessary `list`.

File: deps/email_template/models.py [hunk 34]
```diff
@@ -24,8 +24,8 @@ def render(self, ctx, mark_safestring=True):
         """
         if mark_safestring:
             ctx = dict(ctx)
-            for key, value in ctx.items():
-                if isinstance(value, basestring):
+            for key, value in list(ctx.items()):
```
Comment (umgelurgel) [inline deps/email_template/models.py]: Unnecessary `list`.

File: deps/email_template/utils.py [hunk 35]
```diff
@@ -11,7 +11,7 @@ def email(code, ctx, mark_safestring=True, **kw):
     email.body = body
     if e.from_email and 'from_email' not in kw:
         kw['from_email'] = e.from_email
-    for key, value in kw.items():
+    for key, value in list(kw.items()):
```
Comment (umgelurgel) [inline deps/email_template/utils.py]: Unnecessary `list`.

File: p3/admin.py [hunk 36]
```diff
@@ -306,7 +306,7 @@ def stats_data(self, request):
                 offset = date - c.conference_start
                 data[tt][offset.days] += 1
 
-            for k, v in data.items():
+            for k, v in list(data.items()):
```
Comment (umgelurgel) [inline p3/admin.py]: Unnecessary `list`.

File: p3/admin.py [hunk 37]
```diff
@@ -610,14 +610,14 @@ def _calc_prices(order_id, items):
                     'name': fname,
                     'price': price,
                 })
-    for oid, items in grouped.items():
+    for oid, items in list(grouped.items()):
```
Comment (umgelurgel) [inline p3/admin.py]: Unnecessary `list`.

File: p3/admin.py [hunk 38]
```diff
@@ -610,14 +610,14 @@ def _calc_prices(order_id, items):
...
         _calc_prices(oid, items)
 
     # after using _calc_prices obtain the prices not found anymore
     # of the ordinary rates, regroup the resulting OrderItem
     # by rate code and new price
     tcp = {}
-    for rows in grouped.values():
+    for rows in list(grouped.values()):
```
Comment (umgelurgel) [inline p3/admin.py]: Unnecessary `list`.

File: p3/dataaccess.py [hunk 39]
```diff
@@ -287,7 +287,7 @@ def tags():
         .values_list('profile_id', flat=True))
     hset = set([(cid, pid) for pid in hprofiles])
     data = ctags()
-    for tag, objects in data.items():
+    for tag, objects in list(data.items()):
```
Comment (umgelurgel) [inline p3/dataaccess.py]: Unnecessary `list`.

File: p3/forms.py [hunk 40]
```diff
@@ -541,7 +541,7 @@ def __init__(self, *args, **kwargs):
         # Moreover each reservation must specify the "period".
 
         # These cases will be handled in custom code of clean
-        for k in self.fields.keys():
+        for k in list(self.fields.keys()):
```
Comment (umgelurgel) [inline p3/forms.py]: Unnecessary `list`.

File: p3/management/commands/create_speaker_coupons.py [hunk 41]
```diff
@@ -53,7 +53,7 @@
 # Coupon prefixes used in the above dictionary
 COUPON_PREFIXES = tuple(prefix
                         for ttype, (prefix, discount)
-                        in TALK_TYPE_DISCOUNTS.items())
+                        in list(TALK_TYPE_DISCOUNTS.items()))
```
Comment (umgelurgel) [inline p3/management/commands/create_speaker_coupons.py]: Unnecessary `list`.

File: p3/management/commands/create_speaker_coupons.py [hunk 42]
```diff
@@ -194,12 +194,12 @@ def handle(self, *args, **options):
 
         # Create coupons
         data = []
-        for sid, entry in speakers.items():
+        for sid, entry in list(speakers.items()):
```
Comment (umgelurgel) [inline p3/management/commands/create_speaker_coupons.py]: Unnecessary `list`.

File: p3/management/commands/create_trainer_training_pass_coupons.py [hunk 43]
```diff
@@ -43,7 +43,7 @@
 # Coupon prefixes used in the above dictionary
 COUPON_PREFIXES = tuple(prefix
                         for ttype, (prefix, discount)
-                        in TALK_TYPE_DISCOUNTS.items())
+                        in list(TALK_TYPE_DISCOUNTS.items()))
```
Comment (umgelurgel) [inline p3/management/commands/create_trainer_training_pass_coupons.py]: Unnecessary `list`.

File: p3/management/commands/create_trainer_training_pass_coupons.py [hunk 44]
```diff
@@ -164,12 +165,12 @@ def handle(self, *args, **options):
 
         # Create coupons
         data = []
-        for sid, entry in speakers.items():
+        for sid, entry in list(speakers.items()):
```
Comment (umgelurgel) [inline p3/management/commands/create_trainer_training_pass_coupons.py]: Unnecessary `list`.

File: p3/management/commands/get_attendees_csv.py [hunk 45]
```diff
@@ -62,11 +62,11 @@ def handle(self, *args, **options):
         )
 
         writer = csv.DictWriter(sys.stdout, columns)
-        writer.writerow(dict(zip(columns, columns)))
+        writer.writerow(dict(list(zip(columns, columns))))
 
         def utf8(d):
             d = dict(d)
-            for k, v in d.items():
+            for k, v in list(d.items()):
```
Comment (umgelurgel) [inline p3/management/commands/get_attendees_csv.py]: Unnecessary `list`.

File: p3/management/commands/speakers_csv.py [hunk 46]
```diff
@@ -36,10 +36,10 @@ def handle(self, *args, **options):
             COL_NAME, COL_TITLE, COL_BIO,
         )
         writer = csv.DictWriter(sys.stdout, columns)
-        writer.writerow(dict(zip(columns, columns)))
+        writer.writerow(dict(list(zip(columns, columns))))
         def utf8(d):
             d = dict(d)
-            for k, v in d.items():
+            for k, v in list(d.items()):
```
Comment (umgelurgel) [inline p3/management/commands/speakers_csv.py]: Unnecessary `list`.

File: p3/management/commands/talk_abstracts.py [hunk 47]
```diff
@@ -67,7 +67,7 @@ def handle(self, *args, **options):
 
         sessions = OrderedDict()
         # Print list of submissions
-        for type_name, session_talks in talks.items():
+        for type_name, session_talks in list(talks.items()):
```
Comment (umgelurgel) [inline p3/management/commands/talk_abstracts.py]: Unnecessary `list`.

File: p3/management/commands/ticket_list.py [hunk 48]
```diff
@@ -106,45 +106,45 @@ def handle(self, *args, **options):
...
                     ]
-                    print '\t'.join(map(str, row))
+                    print('\t'.join(map(str, row)))
         else:
-            for fare, items in non_conference_tickets.items():
-                print '\n\n'
-                print '\t\t\t', fare.code, fare.name.encode('utf-8')
-                print '-' * 80
-                print '\n\n'
+            for fare, items in list(non_conference_tickets.items()):
```
Comment (umgelurgel) [inline p3/management/commands/ticket_list.py]: Unnecessary `list`.

File: p3/templatetags/p3.py [hunk 49]
```diff
@@ -423,9 +423,9 @@ def com_com_registration(user):
     if user.phone and user.phone.startswith('+39'):
         params['ita_mobile'] = user.phone
     params['username'] = name.lower().replace(' ', '').replace('.', '')[:12]
-    for k, v in params.items():
+    for k, v in list(params.items()):
```
Comment (umgelurgel) [inline p3/templatetags/p3.py]: Unnecessary `list`.

File: p3/templatetags/p3.py [hunk 50]
```diff
@@ -531,7 +531,7 @@ def admin_ticketroom_overall_status():
     rooms = {}
     for day in days:
         dst = status[day]
-        for room_type, dst in status[day].items():
+        for room_type, dst in list(status[day].items()):
```
Comment (umgelurgel) [inline p3/templatetags/p3.py]: Unnecessary `list`.

File: p3/templatetags/p3.py [hunk 51]
```diff
@@ -625,7 +625,7 @@ def match(e, exclude_tags=set(('partner0', 'partner1', 'sprint1', 'sprint2', 'sp
         return True
     events = { x['id']:x for x in filter(match, cdataaccess.events(conf=conference)) }
     talks = {}
-    for e in events.values():
+    for e in list(events.values()):
```
Comment (umgelurgel) [inline p3/templatetags/p3.py]: Unnecessary `list`.

File: p3/utils.py [hunk 52]
```diff
@@ -318,7 +318,7 @@ def group_all_talks_by_admin_type(conference, talk_status='accepted'):
         with status `talk_status` separated by administrative type. """
         # Group by admin types
         talks = OrderedDict()
-        for adm_type, type_name in dict(cmodels.TALK_ADMIN_TYPE).items():
+        for adm_type, type_name in list(dict(cmodels.TALK_ADMIN_TYPE).items()):
```
Comment (umgelurgel) [inline p3/utils.py]: Unnecessary `list`.

File: p3/utils.py [hunk 53]
```diff
@@ -332,7 +332,7 @@ def group_all_talks_by_admin_type(conference, talk_status='accepted'):
                        'helpdesk':    ['h_180'],
                       }
 
-        for grp_name, grp_types in type_groups.items():
+        for grp_name, grp_types in list(type_groups.items()):
```
Comment (umgelurgel) [inline p3/utils.py]: Unnecessary `list`.

File: p3/views/live.py [hunk 54]
```diff
@@ -111,7 +111,7 @@ def event_url(event):
             return None
 
     output = {}
-    for track, event in events.items():
+    for track, event in list(events.items()):
```
Comment (umgelurgel) [inline p3/views/live.py]: Unnecessary `list`.

File: p3/views/schedule.py [hunk 55]
```diff
@@ -70,7 +70,7 @@ def _build_timetables(schedules, events=None, partner=None):
             tts.append((row['id'], tt))
 
     if partner:
-        for date, evts in partner.items():
+        for date, evts in list(partner.items()):
```
Comment (umgelurgel) [inline p3/views/schedule.py]: Unnecessary `list`.

**Review by umgelurgel — CHANGES_REQUESTED** [review]
Marked all cases of where an iterator is cast to a list just to be iterated over - forgot to mention it in the previous PR.

File: p3/views/live.py (hunk 54 above)
Comment (fpierfed) [reply p3/views/live.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: p3/utils.py (hunk 53 above)
Comment (fpierfed) [reply p3/utils.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: p3/utils.py (hunk 52 above)
Comment (fpierfed) [reply p3/utils.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: p3/templatetags/p3.py (hunk 51 above)
Comment (fpierfed) [reply p3/templatetags/p3.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: conference/models.py (hunk 17 above)
Comment (fpierfed) [reply conference/models.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: conference/ical.py (hunk 16 above)
Comment (fpierfed) [reply conference/ical.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: conference/ical.py (hunk 15 above)
Comment (fpierfed) [reply conference/ical.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: conference/fares.py (hunk 14 above)
Comment (fpierfed) [reply conference/fares.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

File: assopy/middleware.py (hunk 13 above)
Comment (fpierfed) [reply assopy/middleware.py]: For future reference: find . -type f -name \*.py -exec gawk -i inplace '{print gensub(/for (.+) in list\((.+)\):/, "for \\1 in \\2:", "g");}' {} \;

**Review by umgelurgel — APPROVED** [review]
Awesome!
